### ISSUE TITLE
Unable to import backup from onedrive

### ISSUE THREAD

**Issue by wangxue415** [Tampermonkey/tampermonkey#2284]
As shown in the following figure, the backup cannot be displayed
![image](https://github.com/user-attachments/assets/c397d945-69ba-452f-bcd3-3caa5e60b3c5)

But you can export the backup to OneDrive

![image](https://github.com/user-attachments/assets/5928c681-9a7b-4463-b7e4-7bfc2d66f66a)

I tried clearing the access token from tampermonkey but it didn't work
![image](https://github.com/user-attachments/assets/e27a7831-7dea-4753-aa0c-7ed19ece9faa)

## Specifications

- edge 131.0.2903.86

**Comment by wangxue415** [comment]
![image](https://github.com/user-attachments/assets/2274f57b-c3ed-4c64-b649-c8a044a6ac1f)

**Comment by wangxue415** [comment]
![image](https://github.com/user-attachments/assets/8b1133ab-aebd-4462-80db-7f4dd10f6dfd)

**Comment by wangxue415** [comment]
![image](https://github.com/user-attachments/assets/11ff59d5-c0ce-404d-9473-8cfede23ae74)

**Comment by derjanb** [comment]
This is working fine here using version 131.0.2903.86 (Official build) (64-bit) and Tampermonkey 5.3.2.

What is your Tampermonkey version? Can you please try a [new temporary browser profile](https://www.microsoft.com/en-us/edge/learning-center/how-to-add-new-profiles?form=MA13I2)?

**Comment by wangxue415** [comment]
I tried clearing the configuration file, but still couldn't import it

**Comment by wangxue415** [comment]
The cause of the matter is as follows:
I tried to tamper with the monkey import configuration of OneDrive using Edge on a new device. My account enabled 2FA, and then I clicked on the wrong number while confirming the number. Then the import failed
Afterwards, I also failed to import on all other devices

**Comment by wangxue415** [comment]
I think there is a problem with my Microsoft account

**Comment by N190392** [comment]
Same here, tried on multiple devices.
![image](https://github.com/user-attachments/assets/72dc3f1d-a26c-4828-b2be-95bb756e30d8)

Weird thing is that i can Export successfully to OneDrive..

**Comment by NoNzCode** [comment]
Currently, the version I am using is the same. It cannot export or display backup information.

**Comment by YiyuMeowzZ** [comment]
坚果云的webDAV也用不了

**Comment by derjanb** [comment]
> NutCloud's webDAV is not working either

@YiyuMeowzZ This is another issue. Please open a new issue and [check the options and background page console for errors and failing network requests](https://tmnk.net/faq#Q600) and attach those information.

**Comment by derjanb** [comment]
Note: this seems to be a OneDrive issue: openstyles/stylus/issues/1853

**Comment by xiaofeiTM233** [comment]
I asked Microsoft OneDrive Support, but they seem to be passing the buck
![image](https://github.com/user-attachments/assets/1582c550-310e-4cac-a150-75298c05f05a)
I guess it's them changing the way they use apis without telling developers

**Comment by OreoProMax** [comment]
same here

**Comment by nilxbit** [comment]
same problem

**Comment by nilxbit** [comment]
I found that the correct url should be 
```
https://api.onedrive.com/v1.0/drive/special/approot/children
```
but I don't know why it is
```
https://api.onedrive.com/v1.0/drive/special/approot::/children
```

![Image](https://github.com/user-attachments/assets/9e6fb745-6f51-460f-81ee-9d33b937deba)

![Image](https://github.com/user-attachments/assets/2cca2f2a-9f61-442a-a7ef-145b8e4121d3)

![Image](https://github.com/user-attachments/assets/9ab8e2a9-9082-40c8-a9eb-61ce22193c63)

**Comment by xiaofeiTM233** [comment]
> I found that the correct url should be
> 
> ```
> https://api.onedrive.com/v1.0/drive/special/approot:/:/children
> ```

I try to change it to `https://api.onedrive.com/v1.0/drive/special/approot:/:/children` after I can show my backup normally, that seems to be the root of the problem

**Comment by derjanb** [comment]
This should be fixed at 5.4.6226 ([crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|xpi - still in review)

Chrome/Edge users, please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions`.
Firefox users please install the [BETA version](https://www.tampermonkey.net/index.php?browser=firefox) or check for BETA version updates at `about:addons` **once the review is finished**.

For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.

**Comment by nilxbit** [comment]
It has been fixed, thank you

**Comment by arkM4nKr** [comment]
The Delete API for OneDrive seems to have been modified. Instead of being able to use paths to delete files, only {drive-id} and {item-id} can be used, which has led to Tampermonkey not being able to delete backed up files.

[Example from Microsoft ](https://learn.microsoft.com/en-us/onedrive/developer/rest-api/api/driveitem_delete#http-request)
DELETE /drives/{drive-id}/items/{item-id} 
DELETE /groups/{group-id}/drive/items/{item-id} 
DELETE /me/drive/items/{item-id}
DELETE /sites/{siteId}/drive/items/{itemId} 
DELETE /users/{userId}/drive/items/{itemId}

![Image](https://github.com/user-attachments/assets/e70094d7-93e4-4e19-a6ab-c57e22510cfc)
